src/python/pants/build_graph/address.py

```diff
@@ -388,6 +388,17 @@ def sanitize(s: str) -> str:
         prefix = sanitize(self.spec_path)
         return f"{prefix}{path}{target}{params}{generated}"
 
+    def parametrize(self, parameters: dict[str, str]) -> Address:
```

Comment:
Does mypy accept an instance of `FrozenDict` as `parameters` here? 

I'm used to seeing `Mapping[str, str]` where the arg is immutable.

Reply:
Yea, good point. I'll adjust to `Mapping`.